### PR TITLE
Clarify that cache is not strictly necessary

### LINKED ISSUE #11: What is the effect of not using a proper cache setup?
I'm using the default Django cache setup (that is to say, no cache config at all).

Is it just a performance hit? Or are there circumstances in which the select would not work?

### PR DESCRIPTION
**Pull request by abeforgit**
While it is definitely a good idea to stress the importance of a good cache setup, the current docs make it sound like it won't work at all without, which doesn't seem to be the case. For quick tests and small-scale projects it is good to know that it will run fine without.
Closes #11

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -21,7 +21,7 @@ Add ``django_select`` to your URL root configuration:
         # … other patterns
     ]
 
-Finally make sure you have a persistent cache backend setup (NOT
+While not strictly necessary for django-select to work, for performance reasons it is highly recommended to setup a persistent cache backend (NOT
```
Comment (jieter) [inline docs/index.rst:24]: I suggest using the exact project name (django-select2)

```suggestion
While not strictly necessary for django-select2 to work, for performance reasons it is highly recommended to setup a persistent cache backend (NOT
```

You might consider adhering to the surrounding line length (or at least use "Semantic Line Breaks" https://sembr.org/).

**Comment by solarissmoke** [comment]
I think the assertion that a cache isn't required is false. You can verify this by running this package with the cache entirely disabled (i.e.,g use the Django `DummyCache`) as a backend: it will not work at all. [This bit of code](https://github.com/codingjoe/django-select2/blob/master/django_select2/views.py#L84-L87) will always raise a 404 error because there is never anything in the cache. So instead of saying that a cache is optional, I think the documentation in fact needs to state that *a working cache backend is mandatory*.

There's a more fundamental problem, which is that if things fall out of the cache then the behaviour breaks. The code is not only assuming the presence of a cache, but of a *persistent cache that never culls anything*. This was alluded to [in this issue](https://github.com/applegrew/django-select2/issues/243), where the conversation didn't go so well, but I think the original point is valid: it's very easy to break a widget just because it's cache entry was removed for any reason.

**Comment by jpm92** [comment]
> I think the assertion that a cache isn't required is false. You can verify this by running with the cache entirely disabled (i.e., use the Django `DummyCache`) as a backend: it will not work at all. [This bit of code](https://github.com/codingjoe/django-select2/blob/master/django_select2/views.py#L84-L87) will always raise a 404 error because there is never anything in the cache.
> 
> The code is not only assuming the presence of a cache, but of a _persistent cache that never culls anything_. This was alluded to [in this issue](https://github.com/applegrew/django-select2/issues/243), where the conversation didn't go so well, but I think the original point is valid: it's very easy to break a widget just because it's cache entry was removed for any reason. This is easy to demonstrate with the `LocMemCache` in development - just restarting the Django server while using a widget on a page is sufficient to make it stop working.

Well, I'm using it at the moment with the default django cache and I facing no issues sofar...

File: docs/index.rst (hunk 1 above)
Comment (abeforgit) [reply docs/index.rst]: Good suggestions, applied!

**Comment by abeforgit** [comment]
Im using the default cache (which from the django docs I gather is the LocMemCache), which is also not persistent, with no issues.
Sure maybe it's not the best thing to do and definitely not for a heavy-use production app, but how many of us came here to just get a working select with the least amount of setup for their 5-user demo app?

**Review by codingjoe — APPROVED** [review]
Works for me thanks @abeforgit

**Review by codingjoe — CHANGES_REQUESTED** [review]
Wait, actually it doesn't work for me. The cache needs to be `persistent`, not matter the setup. If you use a single app node setup, of course you don't need a cache that is persistent across mutliple nodes, thus `LocMemCache` would do.

The paragraph was introduced because I got many people reporting problems because they forgot to set up a persistent cache. I know it makes initial set up a little harder, but I don't see a way around it.

**Comment by codingjoe** [comment]
That being said, I welcome changes on the docs and this paragraph too. Maybe you can rewrite it a little be more comprehencive.

**Comment by abeforgit** [comment]
Ah I think that is indeed the source of most of the confusion. Ill rewrite it sometime this week, hopefully tomorrow

**Comment by abeforgit** [comment]
"Tomorrow" turned out to be about 4 months later... funny how that goes...

**Review by codingjoe — APPROVED** [review]
Hi @abeforgit,

Excellent addition to the docs! Good documentation is key. Great contribution.

Best,
Joe

**Pull request merged**